packages/atomic/src/components/atomic-modal/atomic-modal.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  Component,
+  h,
+  State,
+  Prop,
+  Element,
+  Watch,
+  Event,
+  EventEmitter,
+} from '@stencil/core';
+import {
+  Bindings,
+  InitializableComponent,
+  InitializeBindings,
+  DeferUntilRender,
+} from '../../utils/initialization-utils';
+import {getFirstFocusableDescendant} from '../../utils/accessibility-utils';
+import {updateBreakpoints} from '../../utils/replace-breakpoint';
+import {once} from '../../utils/utils';
+
+/**
+ * When the modal is opened, the class `atomic-modal-open` is added to the body, allowing further customization.
+ *
+ * @part backdrop - The transparent backdrop hiding the content behind the modal.
+ * @part container - The modal's outermost container with the outline and background.
+ * @part header-wrapper - The wrapper around the header.
+ * @part header - The header at the top of the modal.
+ * @part header-ruler - The horizontal ruler underneath the header.
+ * @part body-wrapper - The wrapper around the body.
+ * @part body - The body of the modal, between the header and the footer.
+ * @part footer-wrapper - The wrapper with a shadow around the footer.
+ * @part footer - The footer at the bottom of the modal.
+ * @internal
+ */
+@Component({
+  tag: 'atomic-modal',
+  styleUrl: 'atomic-modal.pcss',
+  shadow: true,
+})
+export class AtomicModal implements InitializableComponent {
+  @InitializeBindings() public bindings!: Bindings;
+  @Element() public host!: HTMLElement;
+
+  @State() public error!: Error;
+
+  @Prop({mutable: true}) source?: HTMLElement;
+  @Prop({reflect: true, mutable: true}) isOpen = false;
+  @Prop({mutable: true}) close: () => void = () => (this.isOpen = false);
```

Comment:
close is passed as a prop, since any modal that wraps `atomic-modal` with their own `isOpen` prop needs to set that prop when the `atomic-modal` is closed as a result of clicking the backdrop.